src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngAlgorithm.cs

```diff
@@ -25,6 +25,12 @@ public CngAlgorithm(string algorithm)
             _algorithm = algorithm;
         }
 
+        private CngAlgorithm(string algorithm, ECNamedCurve curve)
+        {
```

Comment:
If we expect these to meet certain conditions, like being non-null, it'd be good to assert that.